technobly/VEXTREME
# Implements #55 High Score (by @TylerBrymer PR #56)

### Problem

This PR is #56 rebased with fixes and improvements

### Solution

- PR #55 rebased on master
- whitespace fixes
- fixes a few pointer/size comparisons
- adds conditional highscore.c logging { #define DEBUG_HIGHSCORE (1) }
- makes high_score_flag readable from parmRam with new RPC ID 16, and speeds memory r/w for highscore
- Dev Mode (cart.bin) and Menu (multicart.bin) should not load a high score

### Steps to Test

Play a bunch of games... if when you lose all lives the high score is set, it should show at the top of the screen when you press reset to get back to the menu.  At that time a new high score record will be saved in `/hs.bin`.  Play a different game, reset, and play the previous game that the high score was saved for, and it should load up in the game when it starts.

### References

Implements #55 
Closes #56
